components/role-mgt/org.wso2.carbon.identity.role.mgt.core/src/main/java/org/wso2/carbon/identity/role/mgt/core/dao/RoleDAOImpl.java

```diff
@@ -855,8 +969,10 @@ protected String getUserIDByName(String name, String tenantDomain) throws Identi
                         String name = resultSet.getString(1);
                         String domain = resultSet.getString(2);
                         if (!disabledDomainName.contains(domain)) {
-                            if (StringUtils.isNotEmpty(domain)) {
+                            if (StringUtils.isNotEmpty(domain) && !StringUtils.equals(primaryDomainName, domain)) {
```

Comment:
StringUtils.isNotEmpty(domain) is redundant here

Reply:
Fixed and updated the PR.